Kudu.Core/Infrastructure/FileSystemHelpers.cs

```diff
@@ -290,6 +293,27 @@ public static void DeleteDirectoryContentsSafe(string path, bool ignoreErrors =
             DeleteDirectoryContentsSafe(Instance.DirectoryInfo.FromDirectoryName(path), ignoreErrors);
         }
 
+        public static bool IsFileSystemReadOnly()
+        {
+            if (TmpFolder.StartsWith("%WEBROOT_PATH%", StringComparison.OrdinalIgnoreCase))
+            {
+                // not able to check, return false since by default we are expecting none readonly file system
+                return false;
+            }
+
+            try
+            {
+                string folder = Path.Combine(TmpFolder, Guid.NewGuid().ToString());
+                CreateDirectory(folder);
+                DeleteDirectorySafe(folder, ignoreErrors: false);
+                return false;
+            }
```

Comment:
will be good not to cached it.  since there will be a cause that file system will change to different mod silently. and we only expect file system will stay in RO mod in a very short period. 

and this method shouldn`t be invoke often, or only use when there is exception.